### PR TITLE
Accumulating spark-bigquery connector metrics

### PR DESCRIPTION
**Pull request by isha97**

### CODE REVIEW

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java [hunk 1]
```diff
@@ -100,4 +101,13 @@ static Optional<String> getGcpRegion() {
   public String getUserAgent() {
     return USER_AGENT + " datasource/" + dataSourceVersion;
   }
+
+  @Override
+  public String getConnectorInfo() {
```
Comment (davidrabinowitz) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java:106]: why this is in JSON format? Can you please add some documentation for that? Alternatively, return a map (which is more generic) and let the client to convert it to JSON

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java [hunk 2]
```diff
@@ -100,4 +101,13 @@ static Optional<String> getGcpRegion() {
...
+    JsonObject jsonObject = new JsonObject();
+    jsonObject.addProperty("connectorVersion", SparkBigQueryUtil.CONNECTOR_VERSION);
```
Comment (davidrabinowitz) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java:108]: Why are the other data points not added here?

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/BigQueryRDDContext.java [hunk 3]
```diff
@@ -84,7 +84,7 @@ public scala.collection.Iterator<InternalRow> compute(Partition split, TaskConte
     SparkEnv.get().metricsSystem().registerSource(sparkMetricsSource);
     BigQueryStorageReadRowsTracer tracer =
         bigQueryTracerFactory.newReadRowsTracer(
-            Joiner.on(",").join(streamNames), sparkMetricsSource);
+            Joiner.on(",").join(streamNames), sparkMetricsSource, Optional.empty());
```
Comment (davidrabinowitz) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/BigQueryRDDContext.java:87]: Add a comment that those metrics are not supported in DSv1 as well as the reasons

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/metrics/SparkBigQueryConnectorMetricsUtils.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                  buildMethod.invoke(
+                      eventBuilderClass
+                          .getDeclaredConstructor(String.class)
+                          .newInstance(connectorVersion)));
+    } catch (ReflectiveOperationException ignored) {
+    }
+  }
+
+  public static String extractDecodedSessionIdFromSessionName(String sessionName) {
+    return sessionName.split("/")[5];
```
Comment (vishalkarve15) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/metrics/SparkBigQueryConnectorMetricsUtils.java:42]: Please add a comment explaining this.

File: spark-bigquery-parent/pom.xml [hunk 5]
```diff
@@ -587,6 +587,7 @@
                             <excludes>
                                 <exclude>com.google.cloud.bigquery.connector.common.**</exclude>
                                 <exclude>com.google.cloud.spark.bigquery.**</exclude>
+                                <exclude>com.google.cloud.spark.events.**</exclude>
```
Comment (vishalkarve15) [inline spark-bigquery-parent/pom.xml:590]: This is not needed.

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/ReadSessionMetrics.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.google.cloud.bigquery.connector.common;
```
[the remark this answers is not in the document]
Comment (isha97) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/ReadSessionMetrics.java:16]: Added

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java (hunk 1 above)
Comment (isha97) [reply spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java:109]: Added the comment for using JsonFormat.

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java (hunk 2 above)
Comment (isha97) [reply spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java:111]: What other data points do we need here?

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**